packages/react-composites/src/composites/MeetingComposite/adapter/AzureCommunicationMeetingAdapter.ts

```diff
@@ -454,6 +463,44 @@ export const createAzureCommunicationMeetingAdapter = async ({
   return new AzureCommunicationMeetingAdapter(callAdapter, chatAdapter);
 };
 
+/**
+ * Arguments for {@link createAzureCommunicationMeetingAdapterFromClient}
+ *
+ * @beta
+ */
+export type AzureCommunicationMeetingAdapterFromClientArgs = {
+  meetingLocator: CallAndChatLocator | TeamsMeetingLinkLocator;
+  callAgent: CallAgent;
+  callClient: StatefulCallClient;
+  chatClient: StatefulChatClient;
+  threadId: string;
+};
+
+/**
+ * Create a {@link MeetingAdapter} using the provided {@link StatefulChatClient} and {@link StatefulCallClient}.
+ *
+ * Useful if you want to keep a reference to {@link StatefulChatClient} and {@link StatefulCallClient}.
+ * Consider using {@link createAzureCommunicationMeetingAdapter} for a simpler API.
+ *
+ * @beta
+ */
+export const createAzureCommunicationMeetingAdapterFromClient = async ({
+  callClient,
+  callAgent,
+  meetingLocator,
+  chatClient,
+  threadId
+}: AzureCommunicationMeetingAdapterFromClientArgs): Promise<MeetingAdapter> => {
+  const locator = isTeamsMeetingLinkLocator(meetingLocator) ? meetingLocator : meetingLocator.callLocator;
+  const createCallAdapterPromise = createAzureCommunicationCallAdapterFromClient(callClient, callAgent, locator);
+
+  const chatThreadClient = await chatClient.getChatThreadClient(threadId);
```

Comment:
Let's ask contoso app to call this line whenever they want to call this function - so parameters will be 
{
  callClient,
  callAgent,
  meetingLocator,
  chatClient,
  chatThreadClient
}
If anyone wants to call createAzureCommunicationMeetingAdapterFromClient, that means they wanna have full access to chatClient and chatThreadClient - which means they will need to have these 2 instances ready
Otherwise we are creating a new chatThreadClient instance in the adapter which could cause some unexpected behavior